### ISSUE TITLE
Question about asynchronous design

### ISSUE THREAD

**Issue by kjkpoi** [microsoft/cpprestsdk#1357]
I'm happy with asynchronous operation with cpprestsdk.

In my thought, the asynchronous design has an advantage in dealing with many requests in the server.
The server doesn't have to wait to process a request before completing the previous requests.

But, asynchronous design can access the same resource from multiple requests.
For example when I use a file to read/write in my cpprestsdk server.
My file stream or class should provide thread-safe design??

**Comment by garethsb-sony** [comment]
Yes, you need to protect access to resources which could be used by simultaneous requests. And make sure not to call blocking `task::get` or `wait` but always use continuations.
